lib/evmone/eof.cpp

```diff
@@ -507,9 +578,47 @@ EOF1Header read_valid_eof1_header(bytes_view container)
 
     header.data_size = section_headers[DATA_SECTION][0];
 
+    header.container_sizes = section_headers[CONTAINER_SECTION];
+    std::vector<uint16_t> container_offsets;
+    auto container_offset = code_offset + header.data_size;
+    for (const auto container_size : header.container_sizes)
+    {
+        header.container_offsets.emplace_back(static_cast<uint16_t>(container_offset));
+        container_offset += container_size;
+    }
+
     return header;
 }
 
+bool append_data_section(bytes& container, bytes_view aux_data)
+{
+    // Validate section header first to make sure we can append data section
+    if (!is_eof_container(container) || get_eof_version(container) != 1)
+        return false;
+
+    const auto section_headers_or_error = validate_eof_headers(container);
+    if (std::holds_alternative<EOFValidationError>(section_headers_or_error))
+        return false;
+
+    const auto header = read_valid_eof1_header(container);
+    const auto new_data_size = header.data_size + aux_data.size();
+    if (new_data_size > std::numeric_limits<uint16_t>::max())
+        return false;
+
+    // Insert position is before first embedded container, or at the end if there're none.
```

Comment:
Hm, so container is last? Need to test this well.

This will really push having metadata at the beginning of the data section.